### DISCUSSION TITLE
Different config in inherited models

### DISCUSSION THREAD

**Discussion by evstratbg** [pydantic/pydantic#5037] [Question] [answered]
Hello, I have the following approach we are using in the api

```
from pydantic import BaseModel, Extra
import datetime


class MyConfig(BaseModel):
    class Config:
        extra = Extra.ignore

class Address(BaseModel):
    street: str


class HouseBase(BaseModel):
    levels: int
    built_at: datetime.date
    address: Address


class CreateHouseModel(HouseBase):
    designer_name: str


class ResponseHouseModel(HouseBase):
    id: int
```

Recently, we decided, that we dont accept extra field in the payloads, thats why we changed base config to `extra = Extra.forbid`. The problem is, we use a lot of base models as parts of request and response, as I tried to show in the example above and we dont need `extra = Extra.forbid` for the responses, `extra = Extra.ignore` is fine. Is there any way to dynamically change that config parameter in every field config recursively or there is a better way to solve this?

In the end of the day, we can create another set of the models just for responses, but would be mostly a copy-paste, wanted t avoid that

**Comment by evstratbg** [comment, marked as answer]
sorry, found the same topic here https://github.com/pydantic/pydantic/discussions/2652

**Comment by dmontagu** [comment]
I'm not sure if it's documented clearly elsewhere, but the "shared base model" approach is described at some point here: https://docs.pydantic.dev/usage/model_config/

I think as part of the next big wave of docs improvements (.. when they come ..), I'd like to add a section on "Common Usage Patterns" or similar showing tips and tricks like this. (It's possible such a section exists and I have forgotten, but I wasn't able to find a clear discussion of this pattern through a cursory search in the docs.)
